include/flash_pm.h

```diff
@@ -0,0 +1,32 @@
+#ifndef ZEPHYR_INCLUDE_FLASH_PM_H_
+#define ZEPHYR_INCLUDE_FLASH_PM_H_
+
+/* Default DT node with fixed-partitions compatibility set */
+#define DEFAULT_DT_FIXED_PARTITIONS DT_INST(0, fixed_partitions)
+
+/* Right now only DT definitions are supported, and required */
+#if DT_NODE_HAS_COMPAT(DEFAULT_DT_FIXED_PARTITIONS, fixed_partitions)
+#define FLASH_AREA_LIST DT_CHILD_NODELABELS(DEFAULT_DT_FIXED_PARTITIONS)
+#else
+#error "Expected default fixed partitions table"
+#endif
+
+/* Number of flash areas */
+#define FLASH_AREA_N UTIL_LIST_LEN(FLASH_AREA_LIST)
+
+#define FLASH_AREA_ID(lbl) __dt_flash_defined_partition__##lbl
```

Comment:
```suggestion
#define FLASH_AREA_ID(lbl) flash_area_id_##lbl
```

Double underscores are reserved for the implementation in C++, so we're trying to avoid using them.

Reply:
ok